homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -70,20 +70,22 @@ async def async_setup_entry(
     coordinator = get_coordinator(hass, entry.entry_id)
 
     @callback
-    def _add_entities() -> None:
+    def _add_entities(devices: set[str] | None = None) -> None:
         """Add devices."""
-        if not coordinator.new_devices:
+        if devices is None:
+            devices = coordinator.new_devices
+        if not devices:
             return
         async_add_entities(
             FritzboxBinarySensor(coordinator, ain, description)
-            for ain in coordinator.new_devices
+            for ain in devices
             for description in BINARY_SENSOR_TYPES
             if description.suitable(coordinator.data.devices[ain])
         )
 
     entry.async_on_unload(coordinator.async_add_listener(_add_entities))
 
-    _add_entities()
+    _add_entities(set(coordinator.data.devices.keys()))
```

Comment:
Copying a dict to a set copies the keys by default. We can remove `.keys()`.

```
Python 3.11.7 (main, Dec  8 2023, 18:56:57) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: test = {"one": 1, "two": 2}

In [2]: set(test)
Out[2]: {'one', 'two'}
```

Reply:
https://github.com/home-assistant/core/pull/105515#discussion_r1427395662